webapp/channels/src/components/advanced_text_editor/use_key_handler.tsx

```diff
@@ -0,0 +1,368 @@
+            return;
+        }
+        messageHistoryIndex.current += 1;
+        handleDraftChange({
+            ...draft,
+            message: messageHistory[messageHistoryIndex.current] || '',
+        });
+    }, [draft, handleDraftChange, messageHistory]);
+
+    const postMsgKeyPress = useCallback((e: React.KeyboardEvent<TextboxElement>) => {
+        const {allowSending, withClosedCodeBlock, ignoreKeyPress, message} = postMessageOnKeyPress(
+            e,
+            draft.message,
+            ctrlSend,
+            codeBlockOnCtrlEnter,
+            postId ? 0 : Date.now(),
+            postId ? 0 : lastChannelSwitchAt.current,
+            caretPosition,
+        );
+
+        if (ignoreKeyPress) {
+            e.preventDefault();
+            e.stopPropagation();
+            return;
+        }
+
+        if (allowSending && isValidPersistentNotifications) {
+            e.persist?.();
+
+            // textboxRef.current?.blur();
+
+            if (withClosedCodeBlock && message) {
+                handleSubmit(e, {...draft, message});
+            } else {
+                handleSubmit(e);
+            }
+
+            // setTimeout(() => {
+            //     focusTextbox();
+            // });
+        }
+
+        emitTypingEvent();
+    }, [draft, ctrlSend, codeBlockOnCtrlEnter, caretPosition, postId, emitTypingEvent, handleSubmit, isValidPersistentNotifications]);
+
+    const handleKeyDown = useCallback((e: React.KeyboardEvent<TextboxElement>) => {
+        const ctrlOrMetaKeyPressed = e.ctrlKey || e.metaKey;
+        const ctrlEnterKeyCombo = (ctrlSend || codeBlockOnCtrlEnter) &&
+            Keyboard.isKeyPressed(e, KeyCodes.ENTER) &&
+            ctrlOrMetaKeyPressed;
+
+        const ctrlKeyCombo = Keyboard.cmdOrCtrlPressed(e) && !e.altKey && !e.shiftKey;
+        const ctrlAltCombo = Keyboard.cmdOrCtrlPressed(e, true) && e.altKey;
+        const shiftAltCombo = !Keyboard.cmdOrCtrlPressed(e) && e.shiftKey && e.altKey;
+        const ctrlShiftCombo = Keyboard.cmdOrCtrlPressed(e, true) && e.shiftKey;
+
+        // fix for FF not capturing the paste without formatting event when using ctrl|cmd + shift + v
+        if (e.key === KeyCodes.V[0] && ctrlOrMetaKeyPressed) {
+            if (e.shiftKey) {
+                isNonFormattedPaste.current = true;
+                timeoutId.current = window.setTimeout(() => {
+                    isNonFormattedPaste.current = false;
+                }, 250);
+            }
+        }
+
+        // listen for line break key combo and insert new line character
+        if (Utils.isUnhandledLineBreakKeyCombo(e)) {
+            handleDraftChange({
+                ...draft,
+                message: Utils.insertLineBreakFromKeyEvent(e.nativeEvent),
+            });
+            return;
+        }
+
+        if (ctrlEnterKeyCombo) {
+            postMsgKeyPress(e);
+            return;
+        }
+
+        if (Keyboard.isKeyPressed(e, KeyCodes.ESCAPE)) {
```

Comment:
I wonder why we do this. I can't think of why we'd want to blur the post textbox, but I can't figure out where it came from since it seems to be from the original AdvancedTextEditor PR

Reply:
I feel it is out of the scope, but we can raise this to UX to see if it is something we want to keep or remove. Keep in mind this also works for the RHS.

Reply:
Going way, way back, this comes from https://github.com/mattermost/mattermost-webapp/pull/9122. We wanted it for the Threads view, but since the component was shared with the RHS, it caused it to apply there too. Your PR to unify the key handlers copied it to the post textbox as well.

Funnily enough, you mention that working in the RHS, but it actually completely loses focus when I hit escape there, and it doesn't go back to any textbox when I start typing again. In the centre channel, it goes back to the post textbox just fine. That's on Hub without your changes though.

That all said, let's leave this as-is then. I don't think this is that big of a deal, and we can always file a ticket if someone complains